libraries/Microsoft.Bot.Builder.AI.QnA/Utils/QnACardBuilder.cs

```diff
@@ -77,48 +77,82 @@ public static IMessageActivity GetSuggestionsCard(List<string> suggestionsList,
         }
 
         /// <summary>
-        /// Get active learning suggestions card.
+        /// Get Card for MultiTurn scenario. (Can be deprected from 4.10.0 release of sdk).
         /// </summary>
         /// <param name="result">Result to be dispalyed as prompts.</param>
-        /// <param name="cardNoMatchText">No match text.</param>
+        /// <param name="cardNoMatchText">No match text.</param>       
         /// <returns>IMessageActivity.</returns>
         public static IMessageActivity GetQnAPromptsCard(QueryResult result, string cardNoMatchText)
+        {
+            return GetQnADefaultResponse(result, true);
+        }
+
+        /// <summary>
+        /// Get Card for Default QnA Maker scenario.
+        /// </summary>
+        /// <param name="result">Result to be dispalyed as prompts.</param>
+        /// <param name="displayPreciseAnswerOnly">renderingchoice.</param>
+        /// <returns>IMessageActivity.</returns>
+        public static IMessageActivity GetQnADefaultResponse(QueryResult result, bool displayPreciseAnswerOnly)
         {
             if (result == null)
             {
                 throw new ArgumentNullException(nameof(result));
             }
 
-            if (cardNoMatchText == null)
-            {
-                throw new ArgumentNullException(nameof(cardNoMatchText));
-            }
-
             var chatActivity = Activity.CreateMessageActivity();
             chatActivity.Text = result.Answer;
-            var buttonList = new List<CardAction>();
+            
+            List<CardAction> buttonList = null;
```

Comment:
Please always initialize variables where you use them. This should just be an empty list. Most of your null-checks then go away and the code becomes much simpler.

Reply:
Updated.

---
In reply to: [430560808](https://github.com/microsoft/botbuilder-dotnet/pull/3935#discussion_r430560808) [](ancestors = 430560808)